tokio-codec/tests/codecs.rs

```diff
@@ -97,6 +97,20 @@ fn lines_decoder_max_length() {
     buf.put("aaabbbc");
     assert!(codec.decode(buf).is_err());
 }
+#[test]
+fn lines_decoder_max_length_underrun() {
+    const MAX_LENGTH: usize = 6;
+
+    let mut codec = LinesCodec::with_max_length(MAX_LENGTH);
+    let buf = &mut BytesMut::new();
+    buf.put("line ");
+    assert_eq!(None, codec.decode(buf).unwrap());
+
+    buf.put("too l");
+    assert_eq!(None, codec.decode(buf).unwrap());
+    buf.put("ong\n");
+    assert_eq!("line too long", codec.decode(buf).unwrap().unwrap());
+}
```

Comment:
Yeah, this works.

I dunno if you've already tried, but I did this by hand when I was reviewing your code, before I mentioned the underrun, and I was getting panics on those index ops into `buf`.

Reply:
Yeah, I fixed the panics in 9b72862d519239db94d3225267c41f15da571bf0. Also, cb8858853b1d71b727bdc5a6adad57fda3892135 added a test that causes that panic.